src/converter/one/jfr/Dictionary.java

```diff
@@ -36,10 +36,21 @@ public int size() {
        return size;
     }
 
-    public void put(long key, T value) {
-        if (key == 0) {
-            throw new IllegalArgumentException("Zero key not allowed");
+    // key[i]==0 is used to signal that the i-th position is unset.
+    // Thus, we map key=key+1, so the user can still use key=0.
+    private static long remapKey(long key) {
+        if (key < 0) {
+            throw new IllegalArgumentException("Negative keys not allowed");
         }
+        return key + 1;
+    }
+
+    private static long remapKeyBack(long key) {
+        return key - 1;
```

Comment:
I understand why you did this, but it hurts me to see a whole method for doing `- 1`

Reply:
Yeah I understand, I think we're doing something meaningful and I'd rather have an additional function than copy-pasting the same comment wherever this operation is used, to explain what is going on. It's only one place tough, let me know if you have a strong opinion about this.

Reply:
Yes, it's just one place, and I prefer inlining `key - 1`.

Alternatively, if you are concerned about finding all places where `1` is added/subtracted, you can replace `1` with `BASE_ID`, but that's not necessary, IMO.